src/State/ProviderInterface.php

```diff
@@ -29,5 +29,5 @@ interface ProviderInterface
      *
      * @return T|Pagination\PartialPaginatorInterface<T>|iterable<T>|null
      */
-    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null;
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|iterable|null;
```

Comment:
I think that this is breaking compatibility

Reply:
But regarding this it's not compatible

https://github.com/api-platform/core/blob/bde59ba121bcf99f4207a34dd5d09da8a6b2c5cc/src/Doctrine/Odm/State/CollectionProvider.php#L41